src/Form/QueueContentsForm.php

```diff
@@ -161,19 +161,24 @@ public function batchFinished(bool $success, array $results) {
     $finished_message = $this->t('Finished with errors.');
 
     if ($success) {
-      $finished_message = $this->formatPlural($results['success'],
+      $finished_message = $this->formatPlural(count($results['success']),
         'One mpx item queued.',
         '@count mpx items queued for updates. Imports will continue in the background.');
     }
 
     $this->messenger()->addMessage($finished_message);
 
-    if ($results['errors'] > 0) {
+    if (count($results['errors']) > 0) {
       $errors_message = $this->formatPlural(
-        $results['errors'],
-        'One mpx item could not be queued. Check logs for details.',
-        '@count mpx items could not be queued for updates. Check logs for details.');
+        count($results['errors']),
+        'The following mpx item could not be queued. Check logs for details.',
+        'The following @count mpx items could not be queued for updates. Check logs for details.');
       $this->messenger()->addError($errors_message);
+
+      /* @var \Lullabot\Mpx\DataService\Media\Media $mpx_item */
+      foreach ($results['errors'] as $mpx_item) {
+        $this->messenger()->addError($mpx_item->getId());
```

Comment:
I wonder if we need to do this all as a single addError call. Otherwise, if someone has multiple tabs open, the session messages could be consumed and displayed without the first sentence as context.

Reply:
I'm fine with doing it in a single call, but I'm not sure that scenario is feasible at all. Keep in mind this is done in the "finished" callback and it's called just once at the very end, not in the operations one between batches.